pandas/core/frame.py

```diff
@@ -3597,7 +3601,9 @@ def transpose(self, *args, copy: bool = False) -> DataFrame:
             new_arr = self.values.T
             if copy:
                 new_arr = new_arr.copy()
-            result = self._constructor(new_arr, index=self.columns, columns=self.index)
+            result = self._constructor(
+                new_arr, index=self.columns, columns=self.index, dtype=new_arr.dtype
+            )
```

Comment:
do we need to pass copy=False here (possibly just to be future-safe)?

Reply:
Yep, I’ll add it in the copy=True pr.

actually, it probably makes sense to go through all usages of constructor and pass copy=False for arrays. Will take a look how many there are